Крестики-нолики_1/TreeNode.cpp

```diff
@@ -0,0 +1,50 @@
+#include "TreeNode.h"
+#include <cassert>
+
+TreeNode::TreeNode(PlayField newField, TreeNode* newTreeNode): parent(newTreeNode), field(newField)
+{
+	if (parent != nullptr)
+		parent->addChild(this);
+}
+
+TreeNode::~TreeNode()
+{
+	for (unsigned int i = 0; i < this->children.size(); i++)
+	{
+		delete children[i];
+	}
+}
+
+bool TreeNode::isTerminal() const
+{
+	return this->field.checkFieldStatus() == PlayField::fsCrossesWin || this->field.checkFieldStatus() == PlayField::fsNoughtsWin || this->field.checkFieldStatus() == PlayField::fsDraw;

```

Comment:
```suggestion
	return field.checkFieldStatus() == PlayField::fsCrossesWin || field.checkFieldStatus() == PlayField::fsNoughtsWin || field.checkFieldStatus() == PlayField::fsDraw;

```
К тому же вызывать трижды checkFieldStatus нехорошо, метод не очень простой, а результат всегда одинаковый. Поэтому целесообразно вызвать его один раз, а потому проверить результат.